tests/unit/test_charm.py

```diff
@@ -1,7 +1,251 @@
-# Copyright 2023 Canonical
+# Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
 
-class TestCharm:
-    def test_dummy_pass(self):
-        assert True
+import tempfile
+import unittest
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from scenario import Container, Context, Model, Mount, Relation, State  # type: ignore[import]
+
+from charm import NSSFOperatorCharm
+
+
+@patch("charm.KubernetesServicePatch", new=Mock)
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.ctx = Context(NSSFOperatorCharm)
+
+    def test_given_fiveg_nrf_relation_not_created_when_pebble_ready_then_status_is_blocked(
+        self,
+    ):
+        container = Container(name="nssf", can_connect=True)
+        state_in = State(
+            containers=[
+                container,
+            ]
+        )
+
+        state_out = self.ctx.run(container.pebble_ready_event, state_in)
+
+        self.assertEqual(
+            state_out.status.unit,
+            BlockedStatus("Waiting for fiveg_nrf relation"),
+        )
+
+    def test_given_nrf_data_not_available_when_pebble_ready_then_status_is_waiting(
+        self,
+    ):
+        container = Container(name="nssf", can_connect=True)
+        nrf_relation = Relation("fiveg_nrf")
+        state_in = State(
+            containers=[container],
+            relations=[nrf_relation],
+        )
+
+        state_out = self.ctx.run(container.pebble_ready_event, state_in)
+
+        self.assertEqual(
+            state_out.status.unit,
+            WaitingStatus("Waiting for NRF data to be available"),
+        )
+        self.assertEqual(
+            state_out.deferred[0].name,
```

Comment:
Following @gruyaume's comment on my last PR regarding multiple asserts in one test. I'd like to clarify when we should avoid them. This test seems reasonable, however, it can be split into one test for the waiting status and another one for the event being deferred. Let me know what you think!

Reply:
In this test the condition is the same for both asserts, the test can fail only if that condition is unmet, regardless of the number of the asserts. (The asserts either will be true together or false together)
I believe it's a bad practice to have multiple asserts if it introduces a test case that can fail for multiple reasons. 
We should to test one thing at a time (and that is true here)
However it might become easier to read if split into two :)
That's my opinion @sanchezfdezjavier, but I'm curios what the others think too :)

Reply:
I agree that tests should only test one thing, but this is different from having only one assertion. In this test case, I am validating that we are waiting. We are doing that by notifying the user through the status and deferring the event. We will never split those 2 behaviors.

In later tests, I have one that tests that the configuration is pushed and one that the pebble layer is applied. They are split in 2 tests because they are different things, even if the setup is exactly the same. We could very well move the configuration file handling to a different flow in the future.